File: cpp/step/math/main_2609.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
	int a, b;
	cin >> a >> b;

	int gcd = 1;
	for (int i = 2; i <= min(a, b);) {
		if (a % i == 0 && b % i == 0) {
			a /= i;
			b /= i;
			gcd *= i;
		} else
			i++;
	}
	cout << gcd << '\n' << gcd * a * b << '\n';
}

File: kotlin/step/math/main_2609.kt
import kotlin.math.min

fun main() {
    val (numA, numB) = readLine()!!.split(' ').map { it.toInt() }
    var a = numA
    var b = numB
    var gcd = 1
    var i = 2
    while (i <= min(a, b)){
        if (a % i == 0 && b % i == 0){
            gcd *= i
            a /= i
            b /= i
        } else {
            i++
        }
    }
    println(gcd)
    println(numA / gcd * numB)
}